### ISSUE TITLE
"Object reference not set to an instance of an object" on PostAsync with a ThermostatUpdateRequest containing a SetHoldFunction. 

### ISSUE THREAD

**Issue by spencerwi** [i8beef/Ecobee#12]
I have this code:

```cs
public async function SetHoldTemp(intnewTemp) 
{
                var request = new ThermostatUpdateRequest
                {
                    Selection = new I8Beef.Ecobee.Protocol.Objects.Selection
                    {
                        SelectionType = "registered",
                        SelectionMatch = ""
                    }
                };
                request.Functions.Add(
                    new I8Beef.Ecobee.Protocol.Functions.SetHoldFunction
                    {
                        Type = "setHold",
                        Params = new I8Beef.Ecobee.Protocol.Functions.SetHoldParams
                        {
                            CoolHoldTemp = newTemp,
                            HeatHoldTemp = newTemp,
                            HoldType = "nextTransition"
                        }
                    }
                );
                var response = await this.ecobeeClient.PostAsync<ThermostatUpdateRequest, Response>(request);
                return response;
}
```

I've tried a variety of "shapes" for my `ThermostatUpdateRequest` object, but the same result comes back every time: "Object reference not set to an instance of an object." I've tried even directly copy-pasting both from your "EcobeeClientTest" and your MQTT-to-Ecobee bridge project (both of which have any code paths with a "set hold" commented out, which I suspect may be because this codepath is just broken), and that fails with the same cause too.

Have you been able to successfully set a temperature hold using this lib?

**Comment by i8beef** [comment]
I think I probably just plain forgot to finish the implementation for that... I remember it having some issue that I was going to get back to and then, life happens and I apparently never got back to it. I'll take a look this week and see if I can get it working though, thanks for pointing it out.

**Comment by i8beef** [comment]
Actually, your test code there isn't instantiating the `response.Functions` list before trying to add to it.  Are you sure it doesn't work when that's fixed?

**Comment by i8beef** [comment]
FYI I updated one limitation on the allowable arguments in the case you specify here, and I have it working in the HomeAutio.Mqtt.Ecobee implementation now. You can copy that as a known good working setup.

**Comment by spencerwi** [comment]
Awesome, thanks for the quick response! 

I did think that the not-yet-initialized state of the Functions list was odd, but that was the most recent thing I'd tried (previously I was creating a list and passing it in, but then when that didn't work I went with the approach you see above, since that's what I saw in the HomeAutio.Mqtt.Ecobee project at the time).

I can confirm that the latest version of this lib now works for setting holds -- thanks again for the quick turnaround!
